### DISCUSSION TITLE
Change field name with select

### DISCUSSION THREAD

**Discussion by sehmaschine** [prisma/prisma#4419] [Q&A] [answered]
I'm trying to rename some fields when using select and it doesn't seem to work ... I'm pretty sure that I'm missing something very basic here ...

This is my current query:

```ts
const data = await prisma.data_riskscores.findMany({
  select: {
    id: true,
    sections: true,
    configuration_country: {
      select: {
        id: true,
        name: true
      }
    }
  }
})
```

Now I'd like to rename `sections` and `configuration_country`, because the data should not necessarily reflect the naming scheme of the database (IMO these are very different usecases). I was thinking about something like this, but my question is ... how do I change the field names?

```ts
const data = await prisma.data_riskscores.findMany({
  select: {
    id: true,
    results: sections,
    configuration_country: {
      as: country,
      select: {
        id: true,
        name: true
      }
    }
  }
})
```

**Comment by nikolasburk** [comment, marked as answer]
Hey @sehmaschine, unfortunately I don't think this is possible off the bat with TypeScript. 

However, if you care about the naming of this particular field in your code and don't want to use the name of the DB column, you can use [`@map`](https://www.prisma.io/docs/concepts/components/prisma-client/configuring-the-prisma-client-api#using-map-and-map-to-rename-fields-and-models-in-the-prisma-client-api) to give it a different name in your Prisma Client API. If it's a relation field, you can just [manually rename it](https://www.prisma.io/docs/concepts/components/prisma-client/configuring-the-prisma-client-api#renaming-relation-fields).

Without knowing your schema, it could be be something like this:

```prisma
model RiskScore {
  country ConfigurationCountry 

  @@map("risk_scores")
}

model ConfigurationCountry {
  riskScore Riskscore

  @@map("configuration_countries")
}
```

**Reply by sehmaschine** [reply]
Thanks @nikolasburk. That worked – although (IMO) it's a bit of a detour.

Changing field names within the select statement would be nicer. Coming from raw SQL this seems to be straightforward. But I have no idea what´s going on here in the background, so maybe it´s not as easy with Prisma. However, maybe this is possible in the future. Definitely not a dealbreaker for now.

**Reply by nikolasburk** [reply]
Yeah, agree! Would be interesting to be able to do that at the API level indeed, feel free to[ open a feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) if you'd like to see this in the future 🙂

**Reply by sehmaschine** [reply]
#4433

**Reply by Rojith1999** [reply]
Yeah it's difficult to do in prisma alternative options prisma provide allow you to use raw query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
